src/components/slider/slider-comparator.vue

```diff
@@ -0,0 +1,192 @@
+
+let isDragging = false
+let olLayerPrerenderEvent: EventsKey
+let olLayerPostrenderEvent: EventsKey
+let mapWrapperElement: HTMLElement
+
+statePersistorSliderComparatorService.bootstrap()
+
+watch([sliderTopLayer, sliderActive], ([topLayer, isActive], [oldTopLayer]) => {
+  if (topLayer && isActive) {
+    // Check if top layer has changed
+    // allways unbind prerender/postrender events in this case for the previous top layer
+    if (topLayer !== oldTopLayer) {
+      deactivate()
+    }
+
+    activate()
+  } else {
+    deactivate()
+  }
+
+  olMap.value?.render()
+})
+
+watch(sliderOffset, sliderOffset => {
+  styleObject.left = `${sliderOffset}px`
+  olMap.value?.render()
+})
+
+function activate() {
+  const olLayer = openLayers.getLayerFromCache(sliderTopLayer.value)
+
+  olLayerPrerenderEvent = olLayer.on(
+    <EventTypes>EventType.PRERENDER,
+    function (event) {
+      const ctx = <CanvasRenderingContext2D>(<RenderEvent>event).context
+      const mapSize = olMap.value?.getSize()!
+      const width = sliderOffset.value + sliderElement.value!.offsetWidth / 2
+      const tl = getRenderPixel(<RenderEvent>event, [width, 0])
+      const tr = getRenderPixel(<RenderEvent>event, [mapSize[0], 0])
+      const bl = getRenderPixel(<RenderEvent>event, [width, mapSize[1]])
+      const br = getRenderPixel(<RenderEvent>event, mapSize)
+
+      ctx.save()
+      ctx.beginPath()
+      ctx.moveTo(tl[0], tl[1])
+      ctx.lineTo(bl[0], bl[1])
+      ctx.lineTo(br[0], br[1])
+      ctx.lineTo(tr[0], tr[1])
+      ctx.closePath()
+      ctx.clip()
+    }
+  )
+
+  olLayerPostrenderEvent = olLayer.on(
+    <EventTypes>EventType.POSTRENDER,
+    function (e) {
+      const event = <RenderEvent>e
+      const ctx = <CanvasRenderingContext2D>event.context
+      ctx.restore()
+    }
+  )
+}
+
+function deactivate() {
+  unByKey([olLayerPrerenderEvent, olLayerPostrenderEvent])
+}
+
+function moveSplitBar(offsetLeft: number) {
+  const mapSize = olMap.value?.getSize()!
+  const newRatio =
+    (offsetLeft + sliderElement.value!.offsetWidth / 2) / mapSize[0]
+
+  sliderStore.setRatio(newRatio)
+}
+
+function onMouseDown() {
+  isDragging = true
+}
+
+function onMouseMove(payload: MouseEvent) {
+  if (!isDragging) {
+    return
+  }
+
+  const mapContainerOffsetLeft = mapWrapperElement.offsetLeft!
+
+  moveSplitBar(payload.clientX - mapContainerOffsetLeft)
+}
+
+function onMouseUp() {
+  isDragging = false
+}
+
+function onKeyDownRight() {
+  moveSplitBar(sliderElement.value!.offsetLeft + DEFAULT_STEP_ONKEYDOWN)
+}
+
+function onKeyDownLeft() {
+  moveSplitBar(sliderElement.value!.offsetLeft - DEFAULT_STEP_ONKEYDOWN)
+}
+
+function onKeyDownEsc() {
+  sliderStore.toggleSlider()
+}
+
+onMounted(() => {
+  // Keep ".map-wrapper" for v3 compatibility
+  mapWrapperElement = olMap.value
+    ?.getTargetElement()
+    ?.closest('.map-wrapper') as HTMLElement
+
+  document.addEventListener('mousemove', onMouseMove)
```

Comment:
why not remove event listeners on mouseup as done in old version ? What is the performance penalty ?

Reply:
I have added a mouseUp handler to remove event listeners on mouse up. Can you test this improvement please? I feel it is slightly faster.